### ISSUE TITLE
Errors in stub files silently swallowed for PEP-561 stub-only packages

### ISSUE THREAD

**Issue by aldanor** [python/mypy#8002]
If your `foo-stubs/__init__.pyi` (which is a stubs-only PEP-561-compliant package for `foo`) has some errors, e.g.:

```python
def foo(x: Union[int, str]) -> None: ...
```

then any downstream code you typecheck seems to magically pass somehow (?..), no errors are reported which may be extremely misleading, especially when developing and testing such stub packages.

If you rename `foo-stubs` to just `./foo` (and put it somewhere on `MYPYPATH`), then (when checking downstream code) you instantly get:

```python
__init__.pyi:1: error: Name 'Union' is not defined
__init__.pyi:1: note: Did you forget to import it from "typing"? (Suggestion: "from typing import Union")
```

I'm not sure whether it's a known issue?

**Comment by ilevkivskyi** [comment]
`--no-silence-site-packages`?

**Comment by aldanor** [comment]
But why is it a default for stub-only packages? That is somewhat baffling...

Also, with `--no-silence-site-packages`, I seem to get a million of errors like this from mypy itself:

```
In member "exec_module" of class "InspectLoader":
env/lib/python3.6/site-packages/mypy/typeshed/stdlib/3/importlib/abc.pyi:34: error:
Name 'types.ModuleType' is not defined  [name-defined]
        def exec_module(self, module: types.ModuleType) -> None: ...
```

**Comment by ethanhs** [comment]
What version of mypy are you running? 

Also the default was chosen because there was concern that there would be hard to silence errors in 3rd party stubs.

FWIW, I am beginning to think we should reconsider this decision. I think errors should almost always be opted-out of, rather than opt in (to reduce surprise errors/misleading information). On top of that, if there is an error, one can silence it for that particular library pretty easily via config file.

I'm curious what others think, however.

**Comment by ilevkivskyi** [comment]
> But why is it a default for stub-only packages?

Because library/stub developers are probably around 1% of library/stub users, and the former are assumed to be more advanced in typing matter. Although it is indeed possible to selectively silence errors from stubs, bit this would create a major pain and will make a barrier to adoption of stub packages by _users_.

I would propose to just close this and move on.
